src/UI/Components/Diagram/actions.ts

```diff
@@ -227,12 +231,12 @@ export function appendInstance(
   //map through relatedInstances and either append them or connect to them
   serviceWithReferences.relatedInstances.forEach((relatedInstance) => {
     const isInstanceMain = false;
-    const cellAdded = graph.getCell(relatedInstance.instance.data.id);
+    const cellAdded = graph.getCell(relatedInstance.id);
     if (!cellAdded) {
       appendInstance(
         paper,
         graph,
-        relatedInstance,
+        { instance: relatedInstance, relatedInstances: [] },
```

Comment:
hmm, this is a little confusing, can you add a comment about the reasoning?

Reply:
changed typing and that resolved the issue there